File: app/src/test/java/com/lunesu/pengchauferry/FerryPierTest.kt
package com.lunesu.pengchauferry

import org.junit.Assert.assertEquals
import org.junit.Test

class FerryPierTest {
    @Test
    fun testName() {
        assertEquals("PengChau", FerryPier.PengChau.name)
    }

    @Test
    fun testToString() {
        assertEquals("PengChau", FerryPier.PengChau.toString())
    }

    @Test
    fun testValueOf() {
        assertEquals(FerryPier.PengChau, FerryPier.valueOf("PengChau"))
    }

    @Test
    fun testNearest() {
        assertEquals(FerryPier.PengChau, FerryPier.findNearest(22.279, 114.046))
        assertEquals(FerryPier.Central, FerryPier.findNearest(22.286, 114.161))
    }
}

File: PengChauFerryTests/FerryPierTest.swift
//
//  FerryPierTest.swift
//  PengChauFerryTests
//
//  Created by Lionello Lunesu on 8/3/2020.
//  Copyright © 2020 Lionello Lunesu. All rights reserved.
//

import XCTest
@testable import PengChauFerry

class FerryPierTest: XCTestCase {

    func testName() {
        XCTAssertEqual("PengChau", FerryPier.PengChau.rawValue)
    }

    func testToString() {
        XCTAssertEqual("PengChau", FerryPier.PengChau.toString())
    }

    func testValueOf() {
        XCTAssertEqual(FerryPier.PengChau, FerryPier.valueOf("PengChau"))
    }

    func testNearest() {
        XCTAssertEqual(FerryPier.PengChau, FerryPier.findNearest(latitude: 22.279, longitude: 114.046))
        XCTAssertEqual(FerryPier.Central, FerryPier.findNearest(latitude: 22.286, longitude: 114.161))
    }

    func testNearestCustom() {
        XCTAssertEqual(FerryPier.MuiWo, FerryPier.findNearest(latitude: 22.279, longitude: 114.046, piers: [.MuiWo]))
    }
}
